.orchestra/ci/ci-run.sh

```diff
@@ -129,29 +132,16 @@ fi
 
 REMOTE="$(git remote get-url origin | sed 's|^\([^:]*:\)\([^/]\)|\1/\2|')"
 GITLAB_ROOT="$(dirname "$(dirname "$REMOTE")")"
-echo "${BASE_USER_OPTIONS_YML//\%GITLAB_ROOT\%/$GITLAB_ROOT}" > "$USER_OPTIONS"
-
-# Register target components
-if test -n "$TARGET_COMPONENTS_URL"; then
-    # Add components by repository URL
-    for TARGET_COMPONENT_URL in $TARGET_COMPONENTS_URL; do
-        NEW_COMPONENT="$(orc components --repository-url "$TARGET_COMPONENT_URL" \
-                         | grep '^Component' \
-                         | cut -d' ' -f2)"
-        if test -z "$NEW_COMPONENT"; then
-            log "Warning: ignoring URL $TARGET_COMPONENT_URL since it doesn't match any component"
-        else
-            TARGET_COMPONENTS="$NEW_COMPONENT $TARGET_COMPONENTS"
-        fi
-    done
-fi
 
-log "Target components: $TARGET_COMPONENTS"
+# Replace placeholders in provided user_options.yml with actual values:
+# - %GITLAB_ROOT% -> $GITLAB_ROOT
+# - %PRIVATE_SOURCES_PLACEHOLDER% - Private sources namespace with internal HTTP clone token
+# - %PRIVATE_BIN_ARCHIVES_PLACEHOLDER% - Private binary archives repo with internal HTTP clone token
+USER_OPTIONS_YML="${BASE_USER_OPTIONS_YML//\%GITLAB_ROOT\%/$GITLAB_ROOT}"
+USER_OPTIONS_YML="${USER_OPTIONS_YML//\%PRIVATE_SOURCES_PLACEHOLDER\%/$PRIVATE_SOURCES_CLONE_URL}"
+USER_OPTIONS_YML="${USER_OPTIONS_YML//\%PRIVATE_BIN_ARCHIVES_PLACEHOLDER\%/$PRIVATE_BIN_ARCHIVES_CLONE_URL}"
```

Comment:
This looks terrible. Can we have a `bash` function with a decent name and hiding the complexity of the `bash` syntax?